File: 0844-backspace-string-compare/0844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        stack_s, stack_t = [], []
        
        for c in s:
            if c == '#':
                if stack_s:
                    stack_s.pop()
            else:
                stack_s.append(c)
        
        for c in t:
            if c == '#':
                if stack_t:
                    stack_t.pop()
            else:
                stack_t.append(c)
        
        return "".join(stack_s) == "".join(stack_t)

File: 0844-backspace-string-compare/0844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        string new_s = "", new_t = "";
        
        for (const char& c: s){
            if (c == '#'){
                if (!new_s.empty()) new_s.pop_back();
            }
            else{
                new_s.push_back(c);
            }
        }
        
        for (const char& c: t){
            if (c == '#') {
                if (!new_t.empty()) new_t.pop_back();
            }
            else{
                new_t.push_back(c);
            }
        }
        
        return new_s == new_t;
    }
};